nexus/reconfigurator/planning/src/planner.rs

```diff
@@ -2180,4 +2204,421 @@ mod test {
+                .clickhouse_keeper_cluster_membership
+                .insert(*zone_id, membership);
+        }
+        let blueprint5 = Planner::new_based_on(
+            log.clone(),
+            &blueprint4,
+            &input,
+            "test_blueprint5",
+            &collection,
+        )
+        .expect("created planner")
+        .with_rng_seed((TEST_NAME, "bp4"))
+        .plan()
+        .expect("plan");
+
+        let clickhouse_cluster_config4 =
+            blueprint4.clickhouse_cluster_config.as_ref().unwrap();
+        let clickhouse_cluster_config5 =
+            blueprint5.clickhouse_cluster_config.as_ref().unwrap();
+
+        assert_eq!(
+            clickhouse_cluster_config4.generation,
+            clickhouse_cluster_config5.generation
+        );
+        assert_eq!(
+            clickhouse_cluster_config4.max_used_keeper_id,
+            clickhouse_cluster_config5.max_used_keeper_id
+        );
+        assert_eq!(
+            clickhouse_cluster_config4.keepers,
+            clickhouse_cluster_config5.keepers
+        );
+        assert_ne!(
+            clickhouse_cluster_config4
+                .highest_seen_keeper_leader_committed_log_index,
+            clickhouse_cluster_config5
+                .highest_seen_keeper_leader_committed_log_index
+        );
+        assert_eq!(
+            clickhouse_cluster_config5
+                .highest_seen_keeper_leader_committed_log_index,
+            2
+        );
+
+        logctx.cleanup_successful();
+    }
+
+    // Start with an existing clickhouse cluster and expunge a keeper. This
+    // models what will happen after an RSS deployment with clickhouse policy
+    // enabled or an existing system already running a clickhouse cluster.
+    #[test]
+    fn test_expunge_clickhouse_clusters() {
+        static TEST_NAME: &str = "planner_expunge_clickhouse_clusters";
+        let logctx = test_setup_log(TEST_NAME);
+        let log = logctx.log.clone();
+
+        // Use our example system.
+        let (mut collection, input, blueprint1) =
+            example(&log, TEST_NAME, DEFAULT_N_SLEDS);
+
+        let target_keepers = 3;
+        let target_servers = 2;
+
+        // Enable clickhouse clusters via policy
+        let mut input_builder = input.into_builder();
+        input_builder.policy_mut().clickhouse_policy = Some(ClickhousePolicy {
+            deploy_with_standalone: true,
+            target_servers,
+            target_keepers,
+        });
+        let input = input_builder.build();
+
+        // Create a new blueprint to deploy all our clickhouse zones
+        let mut blueprint2 = Planner::new_based_on(
+            log.clone(),
+            &blueprint1,
+            &input,
+            "test_blueprint2",
+            &collection,
+        )
+        .expect("created planner")
+        .with_rng_seed((TEST_NAME, "bp2"))
+        .plan()
+        .expect("plan");
+
+        // We should see zones for 2 clickhouse keepers, and 2 servers created
```

Comment:
```suggestion
        // We should see zones for 3 clickhouse keepers, and 2 servers created
```

Reply:
fixed in [ad1801b](https://github.com/oxidecomputer/omicron/pull/6627/commits/ad1801bc113ffab526bfca3a073615489028124a)